components/cam/src/dynamics/se/stepon.F90

```diff
@@ -288,129 +300,54 @@ subroutine stepon_run2(phys_state, phys_tend, dyn_in, dyn_out )
-                     endif
-                  enddo
-               end do
-            end do
-         end do
-
-!$omp parallel do private(k, j, i, ic, dp_tmp)
-         do k=1,nlev
-          do ic=1,pcnst
-            do j=1,np
-               do i=1,np
-                  ! make Q consistent now that we have updated ps_v above
-                  ! recompute dp, since ps_v was changed above
-                  dp_tmp = ( hyai(k+1) - hyai(k) )*dyn_ps0 + &
-                       ( hybi(k+1) - hybi(k) )*dyn_in%elem(ie)%state%ps_v(i,j,tl_f)
-                  dyn_in%elem(ie)%state%Q(i,j,k,ic)= &
-                       dyn_in%elem(ie)%state%Qdp(i,j,k,ic,tl_fQdp)/dp_tmp
-               end do
-            end do
-          end do
-         end do
-
-      endif ! if ftype == 2 or == 3 or == 4
+      endif ! if ftype == 2 or == 4
 
       !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
       ! ftype=1:  apply all forcings as an adjustment
       !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
       if (ftype==1) then
          ! apply forcing to state tl_f
          ! requires forward-in-time timestepping, checked in namelist_mod.F90
-!$omp parallel do private(k)
-         do k=1,nlev
-            dp(:,:,k) = ( hyai(k+1) - hyai(k) )*dyn_ps0 + &
-                 ( hybi(k+1) - hybi(k) )*dyn_in%elem(ie)%state%ps_v(:,:,tl_f)
-         enddo
-         do k=1,nlev
-            do j=1,np
-               do i=1,np
 
-                  do ic=1,pcnst
-                     ! back out tendency: Qdp*dtime 
-                     fq = dp(i,j,k)*(  dyn_in%elem(ie)%derived%FQ(i,j,k,ic) - &
-                          dyn_in%elem(ie)%state%Q(i,j,k,ic))
-                     
-                     ! apply forcing to Qdp
-                     dyn_in%elem(ie)%state%Qdp(i,j,k,ic,tl_fQdp) = &
-                          dyn_in%elem(ie)%state%Qdp(i,j,k,ic,tl_fQdp) + fq 
-
-                     ! BEWARE critical region if using OpenMP over k (AAM)
-                     if (ic==1) then
-                        ! force ps_v to conserve mass:  
-                        dyn_in%elem(ie)%state%ps_v(i,j,tl_f)= &
-                             dyn_in%elem(ie)%state%ps_v(i,j,tl_f) + fq
-                     endif
-                  enddo
+!!!!!!!!!!!!!!!!!!!!!!!! same sub as before, this time bfb with ftype1 
+!will be broken due to line Qdp := Qdp+fq , but it is not tested.
+!!!! double check dtime but thats what ftype1 uses
+         call applyCAMforcing_tracers(dyn_in%elem(ie),hvcoord,tl_f,tl_fQdp,dtime,.true.)
 
+         do k=1,nlev
+            do j=1,np
+               do i=1,np       
                   ! force V, T, both timelevels
-                  dyn_in%elem(ie)%state%v(i,j,:,k,tl_f)= &
-                       dyn_in%elem(ie)%state%v(i,j,:,k,tl_f) +  &
-                       dtime*dyn_in%elem(ie)%derived%FM(i,j,:,k)
-                  
+                  do velcomp=1,2
```

Comment:
For ftype1 for dynamics update, should I have called applycamforcing_dynamics from homme? This can be another pr which will include verification of ftype1 + maybe a new test for ftype1.